server/routes/admin.js

```diff
@@ -2,18 +2,62 @@ const express = require('express');
 const router = express.Router();
 const categoryClient = require('../dbClients/categoriesDB');
 const articleClient = require('../dbClients/articlesDB')
+const ObjectId = require('mongodb').ObjectID;
+ 
 
 router.get("/", function (req, res, next) {
   res.send('admin');
 });
 
+router.get('/articles', function (req, res, next) {
+  const callback = (error, articles) => {
+    res.render("admin-list-articles", {
+      articles: articles
+    })
+  }
+  articleClient.findArticles({}, callback);
+});
+
+router.get('/articles/edit/:articleId', (req, res) => {
+  let articleId = req.params.articleId;
+  const callback = (error, article) => {
+    res.render("admin-edit-article", {
+      article: article
+    })
+  }
+  articleClient.findArticleById(articleId, callback);
+})
+
+router.post('/articles/edit/:articleId', function (req, res, next) {
+  let articleId = req.params.articleId;
+  const query = req.body;
+
+  const callback = (error, article) => {
+    res.redirect('/')
+  }
+  articleClient.editArticle(articleId, query, true, callback);
+
+});
+
+router.get('/article/add', (req, res, next) => {
+  res.render('add-articles');
+});
+
+router.put('/article/add', (req, res) => {
```

Comment:
change this to POST (add should be POST)

Reply:
which add plz?
add-article or /article/add

Reply:
I meant `router.post` instead of `router.put`